Simplenote/Classes/SPNoteEditorViewController+Extensions.swift

```diff
@@ -463,17 +463,53 @@ extension SPNoteEditorViewController {
 
     @objc
     func ensureEmptyNoteIsDeleted() {
-        guard let note = currentNote else {
-            return
-        }
-
         guard note.isBlank, noteEditorTextView.text.isEmpty else {
             save()
             return
         }
 
         SPObjectManager.shared().trashNote(note)
-        currentNote = nil
+    }
+
+    @objc
+    private func handleTapOnCreateNewNoteButton() {
+        saveIfNeeded()
```

Comment:
Nipticky: Maybe split this one into 2/3 smaller methods? (I bet you already had that in mind for a follow up, though!!)

Reply:
Thank you, done!